### ISSUE TITLE
AdoptedStreamServerEndpoint does not implement twisted.internet.interfaces.IStreamServerEndpoint

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5878]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)| @ashfall reported|
|-|-|
|Trac ID|trac#5878|
|Type|defect|
|Created|2012-08-17 17:58:31Z|
|Branch|https://github.com/twisted/twisted/tree/interface-test-server-endpoint-5878|

Also, ServerEndpointTestCaseMixin needs to have a test that checks if the endpoints provide interfaces.IStreamServerEndpoint

<details><summary>Searchable metadata</summary>

```
trac-id__5878 5878
type__defect defect
reporter__ashfall ashfall
priority__high high
milestone__ 
branch__branches_interface_test_server_endpoint_5878 branches/interface-test-server-endpoint-5878
branch_author__ashfall ashfall
status__closed closed
resolution__fixed fixed
component__core core
keywords__endpoint endpoint
time__1345226311000000 1345226311000000
changetime__1347189807000000 1347189807000000
version__None None
owner__ashfall ashfall
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_1"></a>|@ashfall commented|
|-|-|

(In [35272]) Branching to 'interface-test-server-endpoint-5878'

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_3"></a>|@ashfall removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/interface-test-server-endpoint-5878)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra set owner to @ashfall|
|-|-|

Thanks. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_5"></a>|@ashfall set status to `closed`|
|-|-|

(In [35560]) Merge interface-test-server-endpoint-5878: AdoptedStreamServerEndpoint now implements twisted.internet.interfaces.IStreamServerEndpoint

Author: ashfall
Reviewer: thijs
Fixes: [#5878](https://github.com/twisted/twisted/issues/5878)
